File: 72. Edit Distance/minDistance.java
// import java.util.*;
class minDistance {
    public int solution(String word1, String word2) {
        int m = word1.length() + 1;
        int n = word2.length() + 1;
        int[][] dp = new int[m][n];
        for (int i = 1; i < m; i++) {
        	dp[i][0] = i;
        }
        for (int j = 1; j < n; j++) {
        	dp[0][j] = j;
        }
        for (int i = 1; i < m; i++) {
        	for (int j = 1; j < n; j++) {
        		if (word1.charAt(i - 1) == word2.charAt(j - 1)) {
        			dp[i][j] = dp[i - 1][j - 1];
        		}
        		else {
        			dp[i][j] = Math.min(dp[i - 1][j] + 1, Math.min(dp[i][j - 1] + 1, dp[i - 1][j - 1] + 1));
        		}
        	}
        }
        return dp[m - 1][n - 1];
    }

    public static void main(String[] argv) {
    	minDistance obj = new minDistance();
    	String word1 = "";
    	String word2 = "";
    	int res = obj.solution(word1, word2);
    	System.out.println(res);
    }
}

File: 72. Edit Distance/minDistance.py
class minDistance:
    def solution(self, word1: str, word2: str) -> int:
        m = len(word1) + 1
        n = len(word2) + 1
        dp = [[0] * n for i in range(0, m)]
        for i in range(1, m):
            dp[i][0] = i
        for j in range(1, n):
            dp[0][j] = j
        for i in range(1, m):
            for j in range(1, n):
                if word1[i - 1] == word2[j - 1]:
                    dp[i][j] = dp[i - 1][j - 1]
                else:
                    dp[i][j] = min(dp[i - 1][j] + 1, dp[i][j - 1] + 1, dp[i - 1][j - 1] + 1)
        return dp[m - 1][n - 1]

word1 = ""
word2 = "execution"
obj = minDistance()
res = obj.solution(word1, word2)
print(res)